Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2020 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.report;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Configuration parameters for general report modules
+ */
+public class GeneralReportSettings implements Serializable {
+    private static final long serialVersionUID = 1L;
+    
+    private List<Long> dataSourcesToProcess;
+    private String reportDirectoryPath;
+    
+    public List<Long> getDataSourcesToProcess() {
```

Comment:
How about getSelectedDataSources/setSelectedDataSources?

Reply:
Not really - selection can also happen via command line args or entries read from a configuration file. I was thinking that process sounded a little heavy-duty for reporting. It's not a big deal though.